src/main/java/bio/terra/common/iam/SamUserFactory.java

```diff
@@ -64,7 +64,8 @@ public SamUser from(BearerToken bearerToken, String samBasePath) {
     } catch (final NullPointerException e) {
       throw new UnauthorizedException(e.getMessage(), e);
     } catch (final ApiException e) {
-      if (e.getCode() == HttpStatus.NOT_FOUND.value()) {
+      if (e.getCode() == HttpStatus.NOT_FOUND.value()
+          || e.getCode() == HttpStatus.FORBIDDEN.value()) {
         throw new UnauthorizedException("User not found", e);
       } else {
         throw new InternalServerErrorException(e);
```

Comment:
It seems odd to have an exception that contains a http status and then use it to map to a different http status. Another approach would be to map the ApiException's status directly to the new exception value
```
      throw new ErrorReportException(e, HttpStatus.valueOf(e.getCode())) {};
```

Reply:
it's a fair point, but according to Core Services team there are reasons for returning a 401 here, and revisiting that is beyond the scope of this PR, so I'm not going to touch it for now.